storage/subscriber.go

```diff
@@ -164,10 +165,29 @@ func (db *store) UpdateSubscriber(s *entities.Subscriber) error {
 }
 
 // DeactivateSubscriber de-activates a subscriber by the given user and email.
-func (db *store) DeactivateSubscriber(userID int64, email string) error {
-	return db.Model(&entities.Subscriber{}).
-		Where("user_id = ? AND email = ?", userID, email).
+func (db *store) DeactivateSubscriber(userID int64, us *entities.UnsubscribeEvent) error {
```

Comment:
The argument in the method does not seem right to me here in the context of "deactivate subscriber". It's better if this method accepts the subscriber that we want to deactivate (or their id), instead of some "event". The unsubscribe event should be an internal thing here.